### ISSUE TITLE
Tests and solution no longer fit the original intent of the lab

### ISSUE THREAD

**Issue by Sillhouette** [learn-co-curriculum/forms-and-basic-associations-rails-lab#23]
I worked with a student on this today and it seems that sometime in the recent past the tests and the solution branch have been refactored to use the `accepts_nested_attributes` rails helper which hasn't yet been taught to the students. There should be no expectation for them to be using this method to complete the lab. The old solution and tests tested for the manual creation of the notes via a `notes_contents=`method as well as manually creating the html input elements on the page. 

Here is my fork of the lab which better follows the intent of the lab in context with the previous readme lesson: https://github.com/Sillhouette/forms-and-basic-associations-rails-lab-v-000

@rrcobb @maxwellbenton

**Comment by lizbur10** [comment]
I also ran into this with a student. He was getting the following error: 
`Capybara::ElementNotFound:
       Unable to find field :song_notes_attributes_0_content that is not disabled`
As Austin said, `accepts_nested_attributes` has not yet been covered at this point in the curriculum. The explanation below appears in the solution code for the Song model, which is an odd way to introduce the concept and an odd way to provide information the student needs to pass the tests.
` This is some high-power Rails magic
  accepts_nested_attributes_for lets us create multiple Notes when creating a
  Song. That's pretty awesome. However we can define a proc (a small in-line
  method) that is passed the attributes that are about to be handed to the
  to-be-created Note and cancel the operation (reject_if) if they are judged
  to be invalid (i.e. the content attribute is empty). Soooo...powerful!
  accepts_nested_attributes_for :notes, :reject_if => proc { |attrs| attrs[:content].blank? }`
The student and I resolved his issue by hard-coding the input id's in his code to match what Capybara was looking for

**Comment by maxwellbenton** [comment]
Hey @Sillhouette and @lizbur10,

So I _think_ what happened was that a solution was created that diverged from the original learning goals of the lab. That solution no longer worked with the existing tests, so the tests were changed to accommodate.

I've reverted the tests to their previous version, adjusted the solution based on your fork @Sillhouette, and updated the readme accordingly. I believe this should resolve the disconnect and the former solution now works with the tests.
